src/test/rpc/NoRippleCheck_test.cpp

```diff
@@ -220,53 +225,58 @@ class NoRippleCheck_test : public beast::unit_test::suite
         }
     }
 
+public:
+    void run ()
+    {
+        testBadInput ();
+        for (auto user : {true, false})
+            for (auto problem : {true, false})
+                testBasic (user, problem);
+    }
+};
+
+class NoRippleCheckLimits_test : public beast::unit_test::suite
+{
     void
-    testLimits(bool admin, beast::temp_dir const& td)
+    testLimits(bool admin)
     {
         testcase << "Check limits in returned data, " <<
             (admin ? "admin" : "non-admin");
 
         using namespace test::jtx;
 
-        //admin : we create and save a ledger
-        //non-admin : we load the ledger
-        auto ledger = td.file ("ledger");
-        if (! admin)
-            assert (
-                boost::filesystem::exists (
-                    boost::filesystem::path{ledger}));
-
-        Env env {*this,
-            admin ?
-                envconfig () :
-                envconfig ([&](std::unique_ptr<Config> cfg)
-                {
-                    cfg = no_admin (std::move(cfg));
-                    cfg->START_LEDGER = ledger;
-                    cfg->START_UP = Config::LOAD_FILE;
-                    cfg->legacy ("database_path", td.path());
-                    return cfg;
-                })};
+        Env env {*this, admin ? envconfig () : envconfig(no_admin)};
 
         auto const alice = Account {"alice"};
-        if (admin)
-        {
-            env.fund (XRP (1000000), alice);
-            env (fset (alice, asfDefaultRipple));
-            env.close ();
+        env.fund (XRP (100000), alice);
+        env (fset (alice, asfDefaultRipple));
+        env.close ();
 
-            for (auto i = 0; i < 500; ++i)
+        for (auto i = 0; i < ripple::RPC::Tuning::noRippleCheck.rmax + 5; ++i)
+        {
+            if (! admin)
             {
-                auto const gw = Account {"gw" + std::to_string(i)};
-                env.fund (XRP (1000), gw);
-                env (trust (alice, gw["USD"](10)));
-                env.close();
+                // endpoint drop prevention. Non admin ports will drop requests
+                // if they are coming too fast, so we manipulate the resource
+                // manager here to reset the enpoint balance (for localhost) if
+                // we get too close to the drop limit.
+                using namespace ripple::Resource;
+                using namespace std::chrono;
+                using namespace beast::IP;
+                auto c = env.app().getResourceManager()
```

Comment:
I don't have a better suggestion, but would be nice to avoid coupling this test to the details of the resource manager.  But given that this is now a manual test, probably not worth it.